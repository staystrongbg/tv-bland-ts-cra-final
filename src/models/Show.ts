export default interface Show {
  id: number;
  url: string;
  name: string;
  type: string;
  language: string;
  genres: string[];
  status: string;
  runtime: number;
  averageRuntime: number;
  premiered: string;
  officialSite: string;
  schedule: ShowSchedule;
  rating: {
    average: number;
  };
  weight: number;
  network: ShowNetwork;
  webChannel: ShowNetwork;
  dvdCountry: string | null;
  externals: ShowExternals;
  image: ShowImage;
  summary: string;
  updated: number;
  _links: ShowLinks;
}
export interface ShowSchedule {
  time: string;
  days: string[];
}

export interface ShowNetwork {
  id: number;
  name: string;
  country: {
    code: string;
    name: string;
    timezone: string;
  };
}
export interface ShowExternals {
  tvrage: number;
  thetvdb: number;
  imdb: string;
}

export interface ShowImage {
  medium: string;
  original: string;
}

export interface ShowLinks {
  self: {
    href: string;
  };
  previousepisode: {
    href: string;
  };
  nextepisode: {
    href: string;
  };
}
