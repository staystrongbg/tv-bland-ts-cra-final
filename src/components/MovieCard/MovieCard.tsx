import { Rating } from 'react-simple-star-rating';
import styles from '../../styles/MovieCard.module.scss';
import { Link } from 'react-router-dom';
import Show from '../../models/Show';
const MovieCard: React.FC<{ data: Show }> = ({ data }) => {
  return (
    <>
      {data ? (
        <div className={styles.showCard}>
          <Link to={`/shows/${data.id}`}>
            {data.image && data.image.medium && (
              <img src={data.image.medium} alt='' />
            )}
          </Link>
          <div className={styles.desc}>
            <div className={styles.rate}>
              <Rating
                readonly
                allowFraction
                size={18}
                initialValue={data?.rating?.average / 2}
              />
            </div>
            <h3>
              {data.name.length > 40
                ? data.name.slice(0, 39) + '...'
                : data.name}
            </h3>
            <p>Runtime: {data.runtime} min</p>
            {/* <p>Airdate: {data.airtime}</p> */}
          </div>
        </div>
      ) : (
        <h3 style={{ color: '#fff' }}>loading...</h3>
      )}
    </>
  );
};

export default MovieCard;
