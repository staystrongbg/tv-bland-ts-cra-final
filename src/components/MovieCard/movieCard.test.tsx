import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ShowsApi from '../../api/shows';
import Show from '../../models/Show';
import MovieCard from './MovieCard';

it('render app', async () => {
  let render, data: Show;

  await ShowsApi.get('1')
    .then((res) => {
      data = res;
      render = renderer
        .create(
          <BrowserRouter>
            <MovieCard data={data} />
          </BrowserRouter>
        )
        .toJSON();
    })
    .catch((error) => console.error(error));

  expect(render).toMatchSnapshot();
});
