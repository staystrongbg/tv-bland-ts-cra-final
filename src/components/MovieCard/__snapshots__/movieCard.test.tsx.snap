// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div
  className="showCard"
>
  <a
    href="/shows/1"
    onClick={[Function]}
  >
    <img
      alt=""
      src="https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
    />
  </a>
  <div
    className="desc"
  >
    <div
      className="rate"
    >
      <span
        className="style-module_starRatingWrap__q-lJC"
        style={
          Object {
            "direction": "ltr",
          }
        }
      >
        <span
          aria-hidden="true"
          className="style-module_simpleStarRating__nWUxf react-simple-star-rating"
          style={
            Object {
              "cursor": "",
            }
          }
        >
          <span
            className="style-module_emptyIcons__Bg-FZ empty-icons"
            style={
              Object {
                "color": "#cccccc",
              }
            }
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
          <span
            className="style-module_fillIcons__6---A filled-icons"
            style={
              Object {
                "color": "#ffbc0b",
                "left": 0,
                "transition": "",
                "width": "65%",
              }
            }
            title="3.25 out of 5"
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={18}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
    </div>
    <h3>
      Under the Dome
    </h3>
    <p>
      Runtime: 
      60
       min
    </p>
  </div>
</div>
`;
